ted_sws/notice_packager/services/notice_packager.py

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/python3
+
+# notice_packager.py
+# Date:  05/03/2022
+# Author: Kolea PLESCO
+# Email: kalean.bl@gmail.com
+
+"""
+This module provides functionalities to generate notice package.
+"""
+
+import base64
+import os.path
+from pathlib import Path
+from tempfile import TemporaryDirectory, NamedTemporaryFile
+from typing import List, Union
+
+from tests.fakes.fake_notice import FakeNotice
+from ted_sws.domain.model.notice import Notice
+from ted_sws.metadata_normaliser.model.metadata import ExtractedMetadata
+from ted_sws.metadata_normaliser.services.xml_manifestation_metadata_extractor import XMLManifestationMetadataExtractor
+from ted_sws.notice_packager.adapters.archiver import ArchiverFactory, ARCHIVE_ZIP_FORMAT, PATH_TYPE
+from ted_sws.notice_packager.adapters.template_generator import TemplateGenerator
+from ted_sws.notice_packager.model.metadata import ACTION_CREATE
+from ted_sws.notice_packager.services.metadata_transformer import MetadataTransformer
+
+ARCHIVE_NAME_FORMAT = "eProcurement_notice_{notice_id}.zip"
+FILE_METS_XML_FORMAT = "{notice_id}-0.mets.xml.dmd.rdf"
+FILE_METS_ACTION_FORMAT = "{notice_id}_mets2{action}.mets.xml"
+FILE_TMD_FORMAT = "techMDID001.tmd.rdf"
+FILE_RDF_FORMAT = "{notice_id}.rdf"
+
+NOTICE_METADATA_TYPE = ExtractedMetadata
+IN_DATA_TYPE = Union[NOTICE_METADATA_TYPE, Notice, str]
+
+
+def __write_template_to_file(file_path, template_generator, template_metadata):
+    __write_to_file(file_path, template_generator(template_metadata))
+
+
+def __write_to_file(file_path, data, mode: str = 'x'):
+    with open(file_path, mode) as file:
+        file.write(data)
+        file.close()
+
+
+def __validated_in_data(in_data: IN_DATA_TYPE) -> NOTICE_METADATA_TYPE:
+    accepted_types = IN_DATA_TYPE.__args__
+    if not isinstance(in_data, accepted_types):
+        raise TypeError('Notice Packager accepts input data of "%s" types only' % accepted_types)
+
+    # here, needed notice_metadata is extracted from provided in_data
+    notice_metadata = None
+    if isinstance(in_data, str):  # notice_id
+        '''
+        if we get notice_id as in_data, 
+        a Notice must be assigned to in_data for next step of the validation flow
+        '''
+        # get Notice from DB
+        notice_id = in_data
+        in_data = FakeNotice(ted_id=notice_id)
+
+    if isinstance(in_data, Notice):  # Notice
+        '''
+        if we get Notice object as in_data, 
+        notice_metadata should be extracted from it
+        '''
+        notice = in_data
+        notice_metadata = XMLManifestationMetadataExtractor(
+            xml_manifestation=notice.xml_manifestation).to_metadata()
+    elif isinstance(in_data, NOTICE_METADATA_TYPE):  # ExtractedMetadata
+        notice_metadata = in_data
+
+    return notice_metadata
+
+
+def create_notice_package(in_data: IN_DATA_TYPE, rdf_content: Union[str, bytes] = None,
```

Comment:
this function is too long. 

It could be broken down into a couple of smaller functions to ease the reading.